src/mscorlib/corefx/System/Globalization/CompareInfo.cs

```diff
@@ -37,6 +37,10 @@ public enum CompareOptions
         OrdinalIgnoreCase = 0x10000000,   // This flag can not be used with other flags.
         StringSort = 0x20000000,   // use string sort method
         Ordinal = 0x40000000,   // This flag can not be used with other flags.
+        
+        // UpperCaseFirst = 0x01000000
```

Comment:
To expose this option from now will require to have a solid solution on Windows. Windows doesn't support this yet. we can try to do the same as System.Xml.Xsl did by manually manipulating the sort keys but this is just a hack and not the proper way to do it.